pkg/controller/storagecluster/reconcile.go

```diff
@@ -590,6 +591,10 @@ func (r *ReconcileStorageCluster) ensureCephConfig(sc *ocsv1.StorageCluster, req
 // ensureCephCluster ensures that a CephCluster resource exists with its Spec in
 // the desired state.
 func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	if len(sc.Spec.MonDataDirHostPath) == 0 && sc.Spec.MonPVCTemplate == nil && len(sc.Spec.StorageDeviceSets) == 0 {
```

Comment:
There's no reason to have this this early and separate from the latter if/else statements. Move this down as the final `else` around line 762

Reply:
because we don't have the logger variable in the scope of the newCephCluster function. And I don't think we should be changing the signature of the newCephCluster function.